examples/intRVFL.py

```diff
@@ -0,0 +1,539 @@
+        "StatlogVehicle",
+        "SteelPlates",
+        "SyntheticControl",
+        "Teaching",
+        "Thyroid",
+        "TicTacToe",
+        "Titanic",
+        "Trains",
+        "Twonorm",
+        "VertebralColumn2Clases",
+        "VertebralColumn3Clases",
+        "WallFollowing",
+        "Waveform",
+        "WaveformNoise",
+        "Wine",
+        "WineQualityRed",
+        "WineQualityWhite",
+        "Yeast",
+        "Zoo",
+    ]
+
+    def __init__(
+        self,
+        model_class: nn.Module,
+        simul: int = 1,
+        batch_size: int = 1,  # DK: The current code will not work for other values. Because "encoding_density_clipped()" assumes only one sample. Do we want to make it more generic?
+        device: torch.device = torch.device("cpu"),
+    ):
+        self.model_class = model_class
+        self.simul = simul
+        self.batch_size = batch_size
+        self.device = device
+
+        # Collects average accuracy for each simulation of each dataset
+        self.accuracy_collection = torch.zeros(
+            len(self.dataset_collection),
+            self.simul,
+            device=self.device,
+        )
+
+    ## Function compute the accuracy for a given model and data
+    def compute_accuracy(self, model, train_ds, test_ds):
+        # Make data loaders
+        train_ld = torch.utils.data.DataLoader(
+            train_ds, batch_size=self.batch_size, shuffle=True
+        )
+        test_ld = torch.utils.data.DataLoader(
+            test_ds, batch_size=self.batch_size, shuffle=True
+        )
+
+        # DK: I think this needs to be somehow connected to transform (callable, optional) that we have for datasets
+        # Used to make min-max normalization of the data
+        min_val = torch.min(train_ds.data, 0).values.to(self.device)
+        max_val = torch.max(train_ds.data, 0).values.to(self.device)
+
+        # Obtain the classifier for the model
+        model.fit(train_ld, min_val, max_val)
+
+        accuracy = torchmetrics.Accuracy()
+        with torch.no_grad():
+            for samples, labels in tqdm(test_ld, desc="Testing"):
+                samples = samples.to(self.device)
+
+                # DK: I think this needs to be somehow connected to transform (callable, optional) that we have for datasets
+                # Normalize
+                samples = (samples - min_val) / (max_val - min_val)
+
+                # Get values for each class
+                outputs = model(samples)
+                # Make prediction
+                # DK: Point for extension. Currently, it assumes implicitly that there is only one vector (prototype) per class, which does not have to be the case in general
+                predictions = torch.argmax(outputs, dim=-1)
+                accuracy.update(predictions.cpu(), labels)
+
+        return accuracy.compute().item()
+
+    def evaluate(self):
+        # For all datasets in the collection
+        for i in range(len(self.dataset_collection)):
```

Comment:
It would be good to modularize this part to allow more flexibility for the end user. I'm thinking something like this generator of datasets:
```python
for train_ds, test_ds in UCIDatasetCollection(root, download):
    # setup data normalization
    # setup data loaders
    # for repeat in repeats
    # instantiate model
    ...
```

The implementation of the `UCIDatasetCollection` would be something like:
```python
def UCIDatasetCollection(root, download):
    # For all datasets in the collection
    for i in range(len(self.dataset_collection)):
        # Fetch the current dataset
        dataset = getattr(torchhd.datasets, self.dataset_collection[i])
    
        # If no separate test dataset available - do 4-fold cross-validation
        if hasattr(dataset, "num_folds"):
            for fold_id in range(dataset.num_folds):
            # Set test and train datasets for the current fold
                       train_ds = dataset(
                           "../data", train=True, download= download, fold=fold_id
                       )
                       test_ds = dataset(
                           "../data", train=False, download= download, fold=fold_id
                       )

               # Case of avaiable test set
                 else:
                     # Set test and train datasets
                     train_ds = dataset("../data", train=True, download=download)
                     test_ds = dataset("../data", train=False, download=download)

        yield train_ds, test_ds
```

This function is straightforward enough that it could be used in many scenarios, in case the rest of our evaluator doesn't work for their setup. We could put it together with the other datasets in `torchhd.dataset.UCIDatasetCollection`

Reply:
Indeed! Added a slightly modified version of the function into "dataset.py".